lib/blocktree/blocktree.go

```
@@ -241,27 +238,31 @@ func (bt *BlockTree) SubBlockchain(start, end Hash) ([]Hash, error) {
 
 }
 
-// deepestLeaf returns the deepest leaf in the block tree.
-func (bt *BlockTree) deepestLeaf() *node {
-	return bt.leaves.deepestLeaf()
+// best returns the best node in the block tree using the fork choice rule.
+func (bt *BlockTree) best() *node {
+	return bt.leaves.bestBlock()
 }
 
-// DeepestBlockHash returns the hash of the deepest block in the blocktree
-// If there is multiple deepest blocks, it returns the one with the earliest arrival time.
-func (bt *BlockTree) DeepestBlockHash() Hash {
+// BestBlockHash returns the hash of the block that is considered "best" based on the
+// fork-choice rule. It returns the head of the chain with the most primary blocks.
+// If there are multiple chains with the same number of primaries, it returns the one
+// with the highest head number.
+// If there are multiple chains with the same number of primaries and the same height,
+// it returns the one with the head block that arrived the earliest.
+func (bt *BlockTree) BestBlockHash() Hash {
 	bt.RLock()
 	defer bt.RUnlock()
 
 	if bt.leaves == nil {
+		// this shouldn't happen
```

should we return an error in this case?